### PR TITLE
[feature](Cloud) Add property storage vault for OlapTable

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -984,6 +987,15 @@ public static String analyzeStoragePolicy(Map<String, String> properties) {
         return storagePolicy;
     }
 
+    public static String analyzeStorageVault(Map<String, String> properties) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:990]: what's this for? it seems unused

File: fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java [hunk 2]
```diff
@@ -145,11 +148,19 @@ protected Partition createPartitionWithIndices(long dbId, OlapTable tbl, long pa
...
             }
+            if (!storageVaultIdSet) {
+                requestBuilder.setStorageVaultName(storageVaultName);
+            }
 
             LOG.info("create tablets, dbId: {}, tableId: {}, tableName: {}, partitionId: {}, partitionName: {}, "
                     + "indexId: {}",
                     dbId, tbl.getId(), tbl.getName(), partitionId, partitionName, indexId);
-            sendCreateTabletsRpc(requestBuilder);
+            // dataProperty.setStorageVaultId(storageVaultId);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java:158]: we have to check the existence when creating a table with storage vault,
it seems that check is missing?

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java (hunk 2 above)
Comment (ByteYue) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java:158]: Currently the existence check logic lies in MS's code, so the rpc procedure already overlaps the check logic.

File: be/src/olap/delta_writer_context.h [hunk 3]
```diff
@@ -43,6 +43,7 @@ struct WriteRequest {
     std::shared_ptr<OlapTableSchemaParam> table_schema_param = nullptr;
     bool is_high_priority = false;
     bool write_file_cache = false;
+    std::string storage_vault_id;
```
Comment (platoneko) [inline be/src/olap/delta_writer_context.h:46]: 内存都用 int64_t 吧

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java (hunk 1 above)
Comment (platoneko) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:990]: return null if no storage vault

**Review by platoneko — CHANGES_REQUESTED** [review]

**Review by platoneko — DISMISSED** [review]

**Review by gavinchou — APPROVED** [review]

File: be/src/cloud/cloud_storage_engine.cpp [hunk 4]
```diff
@@ -189,51 +193,93 @@ Status CloudStorageEngine::start_bg_threads() {
...
         if (!st.ok()) {
             LOG(WARNING) << "failed to refresh object store info. err=" << st;
             continue;
         }
 
         CHECK(!s3_infos.empty()) << "no s3 infos";
-        for (auto& [id, s3_conf] : s3_infos) {
+        for (auto& [id, vault_info] : s3_infos) {
             auto fs = get_filesystem(id);
+            auto s3_conf = std::get<S3Conf>(vault_info);
```
Comment (gavinchou) [inline be/src/cloud/cloud_storage_engine.cpp:276]: unused  var?

**Review by gavinchou — APPROVED** [review]
LGTM

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Comment by ByteYue** [comment]
run buildall

**Review by platoneko — APPROVED** [review]

**Pull request merged**